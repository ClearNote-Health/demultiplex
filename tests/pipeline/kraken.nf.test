nextflow_pipeline {

    name "Test Workflow main.nf - Kraken"
    script "main.nf"
    profile "test_kraken"
    tag "kraken"
    tag "pipeline"

    test("kraken2") {

        when {
            params {
                outdir        = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert workflow.trace.succeeded().size() == 8 },
                { assert snapshot(
                        path("$outputDir/multiqc/multiqc_data/bcl2fastq_lane_counts.txt"),
                        path("$outputDir/multiqc/multiqc_data/fastp_filtered_reads_plot.txt"),
                        path("$outputDir/multiqc/multiqc_data/bcl2fastq_sample_counts.txt"),
                        path("$outputDir/multiqc/multiqc_data/kraken-top-n-plot.txt"),
                    ).match("multiqc") },
                { assert snapshot(
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/L001/Sample1_S1_L001.kraken2.report.txt"),
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/L001/Sample1_S1_L001.fastp.fastq.gz"),
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/L001/Sample1_S1_L001_R1_001.fastq.gz"),
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/L001/Sample1_S1_L001_summary.txt"),
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/L001/Reports/").list(),
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/L001/Stats/").list(),
                    ).match("kraken") }
            )
        }

    }

}
