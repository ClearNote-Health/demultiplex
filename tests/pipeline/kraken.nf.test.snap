{
    "kraken": {
        "content": [
            "Sample1_S1_L001.kraken2.report.txt:md5,726f60beda853ebb9134f6a712c4196b",
            "Sample1_S1_L001.fastp.fastq.gz:md5,64639ff33d4b07d62a9079f88cbbd373",
            "Sample1_S1_L001_R1_001.fastq.gz:md5,0675fb6365322eaafb33c0f8e862b54b",
            "Sample1_S1_L001_summary.txt:md5,510c6b8a26ff8d530950c05b3ca5a1c6",
            [
                [
                    [
                        [
                            [
                                [
                                    "lane.html:md5,794e48287f47a9f22dcb6b6d0c22c3eb",
                                    "laneBarcode.html:md5,2bbdae3ee57151eab520c966597d7438"
                                ],
                                [
                                    
                                ]
                            ]
                        ],
                        [
                            [
                                [
                                    "lane.html:md5,f741870307050dcea79a838f5971770f",
                                    "laneBarcode.html:md5,ffe2e863811c76cb9da27d5d124e0611"
                                ],
                                [
                                    "lane.html:md5,cca97e771d3491dbc8cd3fe389595b09",
                                    "laneBarcode.html:md5,cca97e771d3491dbc8cd3fe389595b09"
                                ]
                            ],
                            [
                                [
                                    "lane.html:md5,c383b0768d9978733d3f5d3b91c15af0",
                                    "laneBarcode.html:md5,48842c23b9a2816aec540177df870968"
                                ],
                                [
                                    
                                ]
                            ]
                        ]
                    ],
                    "Report.css:md5,eb7d3eb68fc1539f411404987246b59b",
                    "index.html:md5,5747c407854ae2c358d0ec201ce622d8",
                    "tree.html:md5,a1b9bf592973ca829ec69ddf888b7e34"
                ]
            ],
            [
                "AdapterTrimming.txt:md5,48ed2b914b1246c0b5d8667525550946",
                "ConversionStats.xml:md5,8fe0f57f3f5d256a0762dba75ac62d05",
                "DemultiplexingStats.xml:md5,2047ff18f5b9107c084de06e9ff943ad",
                "DemuxSummaryF1L1.txt:md5,03e5fd0c1e3079c5f8c7b4d0501b37ff",
                "FastqSummaryF1L1.txt:md5,0c6f2d87ee183b84d1051cde9a5643d1",
                "Stats.json:md5,8e5f038b8aa9e465599d3575f930e604"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-08-05T15:07:49.463457221"
    },
    "software_versions": {
        "content": [
            "{BCL2FASTQ={bcl2fastq=2.20.0.422}, FALCO={falco=1.2.1}, FASTP={fastp=0.23.4}, KRAKEN2={kraken2=2.1.3, pigz=2.8}, MD5SUM={md5sum=8.3}, Workflow={nf-core/demultiplex=v1.5.0dev}}"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-08-01T22:34:15.140488001"
    },
    "multiqc": {
        "content": [
            "bcl2fastq_lane_counts.txt:md5,eed04dae0bae4250ac5f4ab580e32b0b",
            "fastp_filtered_reads_plot.txt:md5,fe76dc8c1650e83e16f547c1fe8a0767",
            "bcl2fastq_sample_counts.txt:md5,6454d154d69ffa563878f5b7fd3d9879",
            "kraken-top-n-plot.txt:md5,84f10e652a1e2b98f96ffbe1a8ffe114"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-08-05T15:07:45.690953174"
    }
}